metadata-ingestion/docs/sources/kafka/kafka.md

```diff
@@ -241,3 +241,358 @@ config:
 ```
 
 The underlying implementation is similar to [dbt meta mapping](https://docs.datahub.com/docs/generated/ingestion/sources/dbt#dbt-meta-automated-mappings), which has more detailed examples that can be used for reference.
+
+### Schema Resolution
+
+DataHub provides comprehensive schema resolution for Kafka topics with multiple fallback strategies. This feature is **completely independent** of data profiling and specifically handles cases where schemas cannot be found in the schema registry.
+
+**Key Points:**
+
+- Schema resolution works by sampling messages from Kafka topics to extract record names and infer schemas
+- It can be enabled/disabled independently of data profiling
+- It's primarily for **metadata extraction** (schema discovery), not data quality analysis
+- Data profiling (covered in the next section) is for **statistical analysis** of message content
+
+#### Multi-Stage Schema Resolution
+
+When DataHub cannot find schemas in the schema registry, it uses a comprehensive multi-stage approach to resolve schemas for Kafka topics:
+
+```yaml
+source:
+  type: kafka
+  config:
+    # ... other config ...
+
+    # Configure comprehensive schema resolution
+    schema_resolution:
+      enabled: true # Enable multi-stage schema resolution (disabled by default)
+      sample_timeout_seconds: 2.0 # Timeout per topic for record name extraction and schema inference
+      sample_strategy: "hybrid" # "earliest", "latest", or "hybrid"
+      max_messages_per_topic: 10 # Maximum messages to sample per topic (default: 10)
```

Comment:
Default is 10 and all examples are in the order of 5-10.
I wonder if those sample sizes may be too small and so lead to anecdotic and not representative schemas.